programs/uxd/src/instructions/redeem_from_router.rs

```diff
@@ -0,0 +1,350 @@
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+
+use crate::error::UxdError;
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::state::mercurial_vault_depository::MercurialVaultDepository;
+use crate::utils::calculate_depositories_redeemable_amount;
+use crate::utils::calculate_depositories_redeemable_amount::DepositoryInfoForRedeemableAmount;
+use crate::utils::calculate_depositories_target_redeemable_amount;
+use crate::utils::calculate_depositories_target_redeemable_amount::DepositoryInfoForTargetRedeemableAmount;
+use crate::utils::validate_redeemable_amount;
+use crate::validate_is_program_frozen;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_LP_TOKEN_VAULT_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+
+#[derive(Accounts)]
+#[instruction(redeemable_amount: u64)]
+pub struct RedeemFromRouter<'info> {
```

Comment:
Nit: `RedeemWithRouter` ? 

We don't exactly redeem from the router, as the router doesn't hold any money. Depositories does.

Reply:
just following the existing naming on this:
- redeem_from_identity_depository
- redeem_from_mercurial_vault_depository

We could name it: `RedeemFromProtocol `if we wanted to make it more gramatically correct?

Reply:
just `mint` and `redeem` would work actually.

Reply:
Yea from/with/to doesn't make much sense in that context, even initially not sure it was the best choice.
Maybe we go with simple Mint and Redeem, without further discrimination (router inferred)

Reply:
Sounds good, changing to mint/redeem in this PR 👌

Reply:
Done!